### DISCUSSION TITLE
Why do the digest files of the releases on the Downloads page start with a blank?

### DISCUSSION THREAD

**Discussion by gingerbeard-man** [openssl/openssl#21629] [Q&A]
With the new releases out yesterday I've once again stumbled across this minor nit:  The SHA1 and SHA256 digest files for the release tarballs contain a single blank preceeding the hex digits.  Is there a particular reason for this?  It was not the case some years ago.

After manually downloading a release I like to perform a quick visual inspection of its digest and the misalignment irritates me ever so slightly:

```
$ cat openssl-3.1.2.tar.gz.sha256 && sha256sum openssl-3.1.2.tar.gz
 a0ce69b8b97ea6a35b96875235aa453b966ba3cba8af2de23657d8b6767d6539
a0ce69b8b97ea6a35b96875235aa453b966ba3cba8af2de23657d8b6767d6539  openssl-3.1.2.tar.gz
```

The tarball comes from a mirror site but I always download the digest and signature files from www.openssl.org.

I still have an old 1.0.2u release lying around where the hex digits _are_ aligned and I can spot at a glance that the digests are identical:

```
$ cat openssl-1.0.2u.tar.gz.sha256 && sha256sum openssl-1.0.2u.tar.gz
ecd0c6ffb493dd06707d38b14bb4d8c2288bb7033735606569d8f90f89669d16
ecd0c6ffb493dd06707d38b14bb4d8c2288bb7033735606569d8f90f89669d16  openssl-1.0.2u.tar.gz
```

**Comment by mattcaswell** [comment]
The release script we were using for 3.0/3.1 that generates these files is different to the one used for 1.1.1 and earlier. It seems that all the 3.0/3.1 releases have this behaviour, but not the old 1.1.1 and earlier script.

Recently (for the last two releases) we started using an even newer single combined script that does all the work for both old versions (1.1.1) and new versions (3.1/3.0).

The old (1.1.1 and earlier) script (written in perl) generated the hashes like this:

https://github.com/openssl/tools/blob/85972a475ef91042e6be0eda4db3202ccc6d0fd2/release-tools/mkrelease.pl#L276-L290

The `\s` in the regex `$sha256hash =~ s/^.*=\s//;` looks significant to this behaviour.

The new script (a shell script) does it like this:

https://github.com/openssl/tools/blob/4307d538d9f8b2f181e2324907fdba547d545858/release-tools/stage-release.sh#L655-L662

So it just looks for the `=` sign and uses everything after that, even it if includes a space

**Reply by slontis** [reply]
Why cant it trim whitespace?
````
$string =~ s/ //g;
````

**Comment by levitte** [comment]
This stems in a misunderstanding of what IFS does.  I thought `I` was for "Input"...
